get_trace_data.py

```diff
@@ -0,0 +1,160 @@
+        prev_long = min_long
+        while prev_long < max_long:
+            cur_long = min(prev_long + section_size, max_long)
+            prev_lat = min_lat
+            while prev_lat < max_lat:
+                cur_lat = min(prev_lat + section_size, max_lat)
+                bbox_sections.append(to_bbox_str(prev_long, prev_lat, cur_long, cur_lat))
+                prev_lat += section_size
+            prev_long += section_size
+
+        pickle.dump(bbox_sections, open(sections_filename, 'wb'))
+
+    return bbox_sections
+
+
+def process_bbox_sections(output_dir: str, output_tmp_dir: str, session: requests.Session, bbox_sections: list[str],
+                          get_trace_data_for_bbox: Callable[[requests.Session, str, any], dict[str, list]],
+                          conf: any) -> None:
+    for bbox in bbox_sections:
+        result_filename = os.path.join(output_dir, bbox + '.pickle')  # The file on disk where we will store trace data
+
+        if os.path.exists(result_filename):  # FIXME
+            print('Seq for bbox={} exists on disk, reading...')
+            try:
+                # TODO: Remove this, next step just needs the trace data on disk
+                trace_data = pickle.load(open(os.path.join(output_dir, bbox + '.pickle'), 'rb'))
+                print('\n\n##################\nPulled from disk seqs for bbox={}'.format(bbox))
+                print('{}\n##################\n\n'.format(trace_data))
+                continue
+            except (OSError, IOError) as e:
+                print('ERROR: bbox={} was marked as completed, but error while pulling data from disk: {}'.format(bbox,
+                                                                                                                  e))
+
+        # We haven't pulled API trace data for this bbox section yet
+        trace_data = get_trace_data_for_bbox(session, bbox, conf)
+
+        # Avoids potential partial write issues by writing to a temp file and then as a final operation, then renaming
+        # to the real location
+        temp_filename = os.path.join(output_tmp_dir, bbox + '.pickle')
+        pickle.dump(trace_data, open(temp_filename, 'wb'))
+        os.rename(temp_filename, result_filename)
+
+
+if __name__ == '__main__':
+    arg_parser = argparse.ArgumentParser()
+    # TODO: Make this optional and do the planet if so?
+    arg_parser.add_argument('--bbox', type=str, help='Filter by the bounding box on the map, given as `min_longitude,'
+                                                     'min_latitude,max_longitude,max_latitude`', required=True)
+    arg_parser.add_argument('--conf', type=str,
+                            help='JSON of configurable settings for this script, e.g. {\"source\":\"mapillary\",'
+                                 '\"mcid\":\"xxx\",\"sequences_per_page\":50,\"skip_if_fewer_images_than\":5}',
+                            required=True)
+    arg_parser.add_argument('--concurrency', type=int,
```

Comment:
i also wonder if this shouldnt just be part of the `conf`? i also think that maybe we should differentiate between conf for pulling data and conf for map matching. but we can get that in a later pr too.

Reply:
Good point. Renamed `--conf` to `--traces_source` to help define its role as a JSON specific to the pulling data config for now. I'll leave concurrency out since we might end up using some threads for map-matching too.